conans/test/unittests/client/conanfile_loader_test.py

```diff
@@ -373,9 +373,43 @@ def append(data):
 
         try:
             sys.path.append(temp)
+            # myconanlogger module must not be cached between multiple conanfiles
```

Comment:
Everything that is not a local import, local to the current conanfile.py is a global import, and shouldn't be a different instance for every conanfile. Otherwise it means that every conanfile.py in the graph would have its own set of private imports of all Python standard library, and of every other Python modules and dependencies that are being imported.

The way it is designed is that the relative, "private" imports of the current conanfile.py are indeed private. But if something is imported from the global Python scope (because it is in ``sys.path``), then it should work the same way as Python, it should be the same instance, not separate. ``myconanlogger`` should be a unique instance, and this is exactly what the test was testing. Because otherwise, every possible Python import that relies on a global state, like a logger, would be broken.

So I am afraid this test cannot be modified, this change could cause quite a headache to a few users.

Reply:
So what would be your suggestion here if we really want to have the local import of pkg1 (instead of reusing the global import)? 
We have a monorepo with many conanfile.py and in a separate folder the python modules which should be exported together when doing `conan export`. 
How can we import the modules without a `sys.path.append`?

Reply:
I am afraid that basically you can't. This is one of the major reasons ``python_requires`` was introduced long time ago.
The problem is the 
```
 from ...common.myfile import MyClass
E           ImportError: attempted relative import with no known parent package
```
Python doesn't let you to import another file in a sibling folder if you are not in a known package. But ``conanfile.py`` are not regular Python packages, so this doesn't apply and doesn't work.

So the options are:
- You have a copy of what you want to have together with the ``conanfile.py`` as a local import and sibling of ``conanfile.py``, and that file is ``exports`` together with the ``conanfile.py``.
- There is some global Python import that is in the Python path all the time (it could be a ``pip`` installed package, for example). Those files are not local to the recipe ``conanfile.py`` and they are not exported a copy with the ``conanfile.py``
- Use ``python_requires`` to have a common shared python code among several packages

Reply:
> You have a copy of what you want to have together with the conanfile.py as a local import and sibling of conanfile.py, and that file is exports together with the conanfile.py.

This is no solution for us since we have stored our modules at a separate place and we cannot move them.

> Use python_requires to have a common shared python code among several packages

This is not possible for us since we work in a monorepo. We want to have reproducible builds, so our `python_requires` must be a full RREV. So `python_requires` is not very useful here since we would have to adapt each conanfile.py every time we make a change at the base class in order to apply the changes. So we will need something to take a base class from local file - and this is exactly what we do here.

What also comes into my mind: We could rename the classes during the export, so that there is no name-clash anymore with the `sys.path` modules. What do you think about this approach?
Unfortunately there is also some bug (Ref: #13362)